tests/test_taskipy.py

```diff
@@ -178,6 +178,23 @@ def test_running_task_arguments_not_passed_to_post_hook(self):
         self.assertNotSubstr(f'the number in posthook is {some_random_number}', stdout)
         self.assertEqual(exit_code, 0)
 
+    def test_running_task_list(self):
+        cwd = self.create_test_dir_from_fixture('project_with_tasks_that_accept_arguments')
```

Comment:
Few things:
First, let's a create a new test case class, e.g. `ListTasksTestCase`. Look at other test case definitions, this should look similar.

Second, for explicitness purposes, let's not use `project_with_tasks_that_accept_arguments`. Instead, we can create other fixtures (under `tests/fixtures` dir). For this specific test, we can call it `project_with_tasks_to_list`. It can even have fewer tasks than the ones in the current fixture.

Third, let's add the following test cases:
1. a test where we pass both `--list` and a task name and see that we ignore the task and not run it
2. a test where we pass task name followed by `--list` and see that we call the task instead of listing
3. a test where we don't have any tasks and see that we print something like "this project has no tasks to list"

I know that this might sound like a lot, but we want to ensure that none of our future changes will break what you're building here today, and the best way to ensure that would be to cover the behavior tightly with tests.